### PR TITLE
Add cloud init to appliance

### PR DESCRIPTION
**Pull request by carbonin**
Adds cloud-init package and some configuration alterations to prevent the default config from preventing root ssh access and to log only to a file, rather than to the console.

### CODE REVIEW

**Comment by kbrock** [comment]
Please test on an appliance before merging.
`cloud-init` has a rough history in this group and would prefer not getting these issues back into the nightly build. people like @jrafanie and @jvlcek may be able to provide you with more information.

Really want `cloud-init` into our appliance. So I'm very excited that you are working on this.

**Comment by carbonin** [comment]
@kbrock 
Tested on RHEV 3.3, RHEV 3.4 and vSphere 5.0.
Confirmed the resolution of the boot time issue that was previously seen with cloud-init installations.
Also found that the issue described here (https://bugzilla.redhat.com/show_bug.cgi?id=1222591) is still happening, but the workaround also still works.

File: kickstarts/base.ks.erb [hunk 1]
```diff
@@ -325,6 +327,15 @@ ln -sf /dev/null /etc/systemd/system/ctrl-alt-del.target
 mkdir /var/log/journal
 systemctl restart systemd-journald
 
+# Alter cloud-init config to allow root ssh access to the appliance
+[[ -s /etc/cloud/cloud.cfg ]] && sed -i "s/^ssh_pwauth:.*$/ssh_pwauth: True/g" /etc/cloud/cloud.cfg
+[[ -s /etc/cloud/cloud.cfg ]] && sed -i "s/^disable_root:.*$/disable_root: false/g" /etc/cloud/cloud.cfg
```
Comment (jvlcek) [inline kickstarts/base.ks.erb]: disable_root had no effect with the older version of cloud-init. Only the ssh_pwauth was needed. Nice to see it's working now!

File: kickstarts/base.ks.erb [hunk 2]
```diff
@@ -325,6 +327,15 @@ ln -sf /dev/null /etc/systemd/system/ctrl-alt-del.target
...
+
+# Alter cloud-init logging config to prevent logging to the console
+[[ -s /etc/cloud/cloud.cfg.d/05_logging.cfg ]] && sed -i "s/handlers=consoleHandler,cloudLogHandler/handlers=cloudLogHandler/g" /etc/cloud/cloud.cfg.d/05_logging.cfg
+[[ -s /etc/cloud/cloud.cfg.d/05_logging.cfg ]] && sed -i "/^ - \[ \*log_base, \*log_syslog \]/d" /etc/cloud/cloud.cfg.d/05_logging.cfg
+[[ -s /etc/cloud/cloud.cfg.d/05_logging.cfg ]] && sed -i "s/^output:.*$/output: {all: '| tee -a \/var\/log\/cloud-init-output\.log \&> \/dev\/null'}/g" /etc/cloud/cloud.cfg.d/05_logging.cfg
+
 # Create a script to initialize appliance on first boot
 cat > /bin/appliance-initialize.sh <<EOF
 #!/bin/sh
```
Comment (jvlcek) [inline kickstarts/base.ks.erb]: @carbonin  This change goes into different files for the downstream build. See the email I sent you about that.

**Comment by jrafanie** [comment]
Nice @carbonin

**Comment by jvlcek** [comment]
@carbonin :+1:

**Comment by jrafanie** [comment]
@carbonin Looks good, let us know when it's ready for review/merge

**Comment by carbonin** [comment]
Just downloading the build with that last change now.  Will give it a quick test and let you know.

**Comment by carbonin** [comment]
That didn't work at all, so I removed the commit.
This is ready @jrafanie

**Comment by jrafanie** [comment]
@carbonin Can you put the BZ URL/trello links in the commit messages?  Are you familiar with rewriting commit messages?  Ping me if you need help.

**Comment by carbonin** [comment]
@jrafanie BZ URL is in the first commit.  Just added the trello links to the last one.
I'm still not sure what to do with https://bugzilla.redhat.com/show_bug.cgi?id=1222591 as there is a workaround, but nothing in this PR has anything directly to do with fixing it.

**Comment by jrafanie** [comment]
Sorry, I meant to say, we usually put the BZ urls in each commit.

**Comment by carbonin** [comment]
Ah got it.  Just fixed them up.  Do you also want the trello links in every commit?

**Comment by jrafanie** [comment]
@carbonin eventually, we'll try to link trello's to PRs so I guess it wouldn't hurt to do it.  If the bot is able to automate some of the workflows in BZ/trello by adding things to commits, it's worth getting into the habit of doing it now I guess.

**Comment by carbonin** [comment]
@jrafanie Yeah, sounds good, updated again.

**Comment by jrafanie** [comment]
:+1:

**Comment by jrafanie** [comment]
Thanks @carbonin.

Note, for future reference https://trello.com/c/whLBjNdu/102-appliance-cloudinit-agent-hangs-on-vsphere and https://trello.com/c/whLBjNdu link to the same place (the latter redirects to the former).  If you rename the trello card, the latter one is the only one that still works so it's better to remove the last part of the URL.

**Pull request merged**